bullet_train/lib/bullet_train/core_ext/callbacks_instrumentation.rb

```diff
@@ -0,0 +1,60 @@
+require_relative "callbacks_instrumentation/backport"
+
+module ActiveSupport::Callbacks
+  ClassMethods.prepend Module.new {
+    def set_callback(name, *filter_list, &block)
+      *, options = normalize_callback_params(filter_list.dup, block)
+      Instrumenter.options = options.merge(caller_locations:)
+      super
+    ensure
+      Instrumenter.options = nil
+    end
+  }
+
+  Callback.prepend Module.new {
+    def compiled
+      @compiled = super.then { Instrumenter.build self, _1 } unless defined?(@compiled)
+      @compiled
+    end
+  }
+
+  class Instrumenter < Data.define(:callback, :callable, :options)
+    singleton_class.attr_accessor :options
+
+    def self.build(callback, callable)
+      options ? new(callback, callable, options) : callable
+    end
+
+    def initialize(callback:, callable:, options:)
+      caller_locations = options.delete(:caller_locations)
+      @location = Rails.backtrace_cleaner.clean(caller_locations.map(&:to_s)).first if caller_locations
+      @full_name = (callback.name == :commit) ? caller_locations.first.base_label : "#{callback.kind}_#{callback.name}"
+      super
+    end
+    attr_reader :options, :location, :full_name
+    delegate :name, :kind, :filter, to: :callback
+
+    def apply(sequence)
+      case kind
+      in :before then sequence.before(self)
+      in :after then sequence.after(self)
+      in :around then sequence.around(self)
```

Comment:
After migrating the database against `main` and switching over to this branch, I got the following error:

![image](https://github.com/bullet-train-co/bullet_train-core/assets/10546292/554f2b5c-3ad5-45b2-b91c-da7f6a322c1d)

Looking at the first commit that you posted in `backport.rb`, I saw that they started using this in `#apply`:

```ruby
callback_sequence.around(@user_callback, @user_conditions)
```

I went ahead and used that and then everything started working for me properly, but I don't know enough about this class to say if that's the proper fix or not.

Reply:
Looks like there's a test failure that's complaining about the same thing: https://github.com/bullet-train-co/bullet_train-core/actions/runs/8115440263/job/22183312092?pr=760#step:7:34

![CleanShot 2024-03-18 at 13 20 11](https://github.com/bullet-train-co/bullet_train-core/assets/58702/556629d8-1cf6-474e-b34a-b2827331e64f)